src/molecules/Modal/ModalHeader.native.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import styled from 'styled-components/native';
+import PropTypes from 'prop-types';
+import Space from '../../atoms/Space';
+import Flex from '../../atoms/Flex';
+import View from '../../atoms/View';
+import Button from '../../atoms/Button';
+import Divider from '../../atoms/Divider';
+
+const FullScreenModalHeader = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[200]};
+  shadow-color: ${(props) => props.theme.colors.primary[930]};
+  shadow-offset: 0px 4px;
+  shadow-opacity: 1;
+  shadow-radius: 15px;
+  elevation: 2;
+`;
+
+const ModalHeader = ({ children, type, onClose }) => {
+  if (type === 'centered') {
+    return (
+      <Space padding={[1.5, 2, 0.5, 2]}>
+        <View>{children}</View>
+      </Space>
+    );
+  }
+  if (type === 'fullscreen') {
+    return (
+      <Flex flexDirection="row" justifyContent="space-between">
+        <Space padding={[1, 2]}>
+          <FullScreenModalHeader>
+            {children}
+            <Button
+              testID="close-button"
+              variant="tertiary"
+              variantColor="shade"
+              icon="close"
+              size="medium"
+              onClick={onClose}
+            />
+          </FullScreenModalHeader>
+        </Space>
+      </Flex>
+    );
+  }
+  if (type === 'bottomsheet') {
+    return (
+      <React.Fragment>
+        <View>{children}</View>
+        <Divider color="shade.920" />
```

Comment:
Again confirm with Nishant. The bottomsheet doesn't shows divider always. It's upto the consumer. Look at the instances of bottomsheet in our app. All those should be moved to this component. So you should decide if this suffice all the use cases in the app as well if not raise it to designers